include/nbl/asset/IPipeline.h

```diff
@@ -223,7 +223,7 @@ class IPipelineBase
 
 			const IShader* shader = nullptr;
 			// A name of the function where the entry point of an shader executable begins. It's often "main" function.
-			std::string_view entryPoint = {};
+			std::string entryPoint = {};
```

Comment:
I want this struct to remain lightweird an "allocator less" so keep it as a `string_view`

note that the spec constant map is a pointer.

Maybe we need the `SShaderSpecInfo` to be templated/behave differently for IGPU and ICPU where the ICPU version doesn't point at the spec_constant_map_t and entrypoint but those are mutable things instead.